app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/directory/ContactsUpdateTest.kt

```diff
@@ -0,0 +1,165 @@
+package com.github.factotum_sdp.factotum.ui.directory
+
+import android.widget.EditText
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.contrib.DrawerActions
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import com.github.factotum_sdp.factotum.MainActivity
+import com.github.factotum_sdp.factotum.R
+import com.github.factotum_sdp.factotum.placeholder.Contact
+import com.github.factotum_sdp.factotum.utils.ContactsUtils
+import junit.framework.TestCase
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+
+class ContactsUpdateTest {
+    @get:Rule
+    var activityRule = ActivityScenarioRule(MainActivity::class.java)
+
+    companion object {
+        private const val nbContacts = 5
+        @BeforeClass
+        @JvmStatic
+        fun setUpDatabase() {
+            val database = ContactsUtils.setEmulatorGet()
+            MainActivity.setDatabase(database)
+        }
+    }
+
+    @Before
+    fun setUp() {
+        ContactsUtils.emptyFirebaseDatabase()
+
+        runBlocking {
+            ContactsUtils.populateDatabase(nbContacts)
+        }
+        Espresso.onView(ViewMatchers.withId(R.id.drawer_layout))
+            .perform(DrawerActions.open())
+        Espresso.onView(ViewMatchers.withId(R.id.directoryFragment))
```

Comment:
You could import the ViewMatchers methods and not specify ViewMatchers each time.